src/modules/commander/state_machine_helper.cpp

```diff
@@ -568,6 +568,8 @@ bool set_nav_state(vehicle_status_s *status, actuator_armed_s *armed, commander_
 
 		} else if (is_armed && check_invalid_pos_nav_state(status, old_failsafe, mavlink_log_pub, status_flags, false, true)) {
 			// nothing to do - everything done in check_invalid_pos_nav_state
+		} else if (status->avoidance_lost) {
+			status->nav_state = vehicle_status_s::NAVIGATION_STATE_AUTO_LOITER;
```

Comment:
Not sure about this failsafe because RTL could be triggered by low battery etc. Should the vehicle loiter or land?